lib/dataset/include/scipp/dataset/slice.h

```diff
@@ -8,6 +8,20 @@
 
 namespace scipp::dataset {
 
+[[nodiscard]] SCIPP_DATASET_EXPORT std::tuple<Dim, scipp::index>
+get_slice_params(const DataArray &data, const Dim dim, const Variable &value);
+
+[[nodiscard]] SCIPP_DATASET_EXPORT std::tuple<Dim, scipp::index, scipp::index>
+get_slice_params(const DataArray &data, const Dim dim, const Variable &begin,
+                 const Variable &end);
+
+[[nodiscard]] SCIPP_DATASET_EXPORT std::tuple<Dim, scipp::index>
+get_slice_params(const Dataset &data, const Dim dim, const Variable &value);
+
+[[nodiscard]] SCIPP_DATASET_EXPORT std::tuple<Dim, scipp::index, scipp::index>
+get_slice_params(const Dataset &data, const Dim dim, const Variable &begin,
+                 const Variable &end);
+
```

Comment:
I am a bit confused why this was added? Can't see where it is used?

Reply:
They are used here: https://github.com/scipp/scipp/blob/f0293e4f68dd795402af0d2259c445cb5f4cd6cb/lib/python/bind_slice_methods.h#L99
and here: https://github.com/scipp/scipp/blob/f0293e4f68dd795402af0d2259c445cb5f4cd6cb/lib/python/bind_slice_methods.h#L156

Reply:
But was there a change related to that?

Reply:
Yes. This is the last case in the PR description. It used to raise
```
scipp._scipp.core.NotFoundError: Expected <scipp.Dict>
  a:<scipp.Variable> (event: 3)      int64  [dimensionless]  [1, 2, 3] to contain b.
```
and now raises 
```
scipp._scipp.core.DimensionError: Invalid slice dimension: 'b': no coordinate for that dimension. Coordinates are (a, )
```

The reason for writing it in this way was to not duplicate the logic of https://github.com/scipp/scipp/blob/f0293e4f68dd795402af0d2259c445cb5f4cd6cb/lib/dataset/slice.cpp#L14-L27 in the bindings.